Cleanup docs about runners (#3875)
hashicorp/waypoint: website/content/docs/internals/architecture.mdx

  enable this functionality.

- **Runner** - A Waypoint runner executes operations such as build,
  deploy, etc. There are two kinds of runners:

>   * **Static Runner** - A static runner is a runner that is long-lived and setup

Reviewer: ```suggestion
  * **Static Runner** - A static runner is a long-lived runner that is set up
```

>     most likely when a team begins using waypoint.

Reviewer: ```suggestion
when a team begins using waypoint.
```

remote operations will be diminished. If no runners exist, jobs will queue.
Operations include build, deploy, release, etc.

If there are no static runners configured in the system or the user has specified

> `-local=true`, then the CLI will spawn runner itself.

Reviewer: ```suggestion
`-local=true`, then the CLI will spawn a runner itself.
```
